src/init/features/turtles/repo.ts

```diff
@@ -0,0 +1,173 @@
+import { cloudbuildOrigin } from "../../../api";
+import { FirebaseError } from "../../../error";
+import * as gcb from "../../../gcp/cloudbuild";
+import { logger } from "../../../logger";
+import * as poller from "../../../operation-poller";
+import * as utils from "../../../utils";
+import { promptOnce } from "../../../prompt";
+
+const gcbPollerOptions: Omit<poller.OperationPollerOptions, "operationResourceName"> = {
+  apiOrigin: cloudbuildOrigin,
+  apiVersion: "v2",
+  masterTimeout: 25 * 60 * 1_000,
+  maxBackoff: 10_000,
+};
+
+function extractRepoSlugFromURI(remoteUri: string): string | undefined {
+  const match = /github.com\/(.+).git/.exec(remoteUri);
+  if (!match) {
+    return undefined;
+  }
+  return match[1];
+}
+
+function generateConnectionId(stackId: string): string {
+  return `turtles-${stackId}-conn`;
+}
+
+function generateRepositoryId(): string | undefined {
+  return `turtles-repo`;
+}
+
+/**
+ * Prompts the user to link their stack to a GitHub repository.
+ */
+export async function linkGitHubRepository(
+  projectId: string,
+  location: string,
+  stackId: string
+): Promise<gcb.Repository> {
+  const connectionId = generateConnectionId(stackId);
+  await getOrCreateConnection(projectId, location, connectionId);
+
+  let remoteUri = await promptRepositoryURI(projectId, location, connectionId);
+  while (remoteUri === "") {
+    await utils.openInBrowser("https://github.com/apps/google-cloud-build/installations/new");
+    await promptOnce({
+      type: "input",
+      message:
+        "Press any key once you have finished configuring your installation's access settings.",
+    });
+    remoteUri = await promptRepositoryURI(projectId, location, connectionId);
```

Comment:
is there some delay between installing the GitHub app and it showing up on Cloud build?

wondering if we should insert some delay here or have some clarification here ("Uh-oh. We still don't have your GitHub installation token, so we are going to open up the installation window again") before we jump back into the infinite loop.